# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/ActorsParent.cpp
# Commit: 96ab87dbe78f
# Full Hash: 96ab87dbe78f92f9523576dd9ba595adbb5996de
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2019-09-11 21:53:06
# Regressor Bug: 1168606
# File Overlap Count: 1
# Description:
#   Bug 1168606 - Added documentation on Cursor data members. r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D41403
# ==============================================================================

diff -r 7738ed9a662e -r 96ab87dbe78f dom/indexedDB/ActorsParent.cpp
--- a/dom/indexedDB/ActorsParent.cpp	Wed Sep 11 12:46:09 2019 +0000
+++ b/dom/indexedDB/ActorsParent.cpp	Wed Sep 11 11:51:22 2019 +0000
@@ -7526,10 +7526,28 @@
   nsCString mContinuePrimaryKeyQuery;
   nsCString mLocale;
 
-  Key mKey;
-  Key mObjectKey;
-  Key mRangeKey;
-  Key mSortKey;
+  // TODO: Apply the renamings suggested below, and also change related
+  // identifiers (e.g. local variables) and literals.
+
+  Key mKey;  ///< The current key, i.e. the key representing the cursor's
+             ///< position (https://w3c.github.io/IndexedDB/#cursor-position).
+             ///<
+             ///< TODO: Rename this to mPosition
+  Key mObjectKey;  ///< The key representing the cursor's object store position
+  ///< (https://w3c.github.io/IndexedDB/#cursor-object-store-position).
+  ///<
+  ///< TODO: Rename this to mObjectStorePosition
+  Key mRangeKey;  ///< If the cursor is based on a key range, the bound in the
+                  ///< direction of iteration (e.g. the upper bound in case of
+                  ///< mDirection == NEXT). If the cursor is based on a key, it
+                  ///< is unset. If mLocale is set, this was converted to
+                  ///< mLocale.
+                  ///<
+                  ///< TODO: Rename this to mLocaleAwareRangeBound
+  Key mSortKey;   ///< If mLocale is set, this is mKey converted to mLocale.
+                  ///< Otherwise, it is unset.
+                  ///<
+                  ///< TODO: Rename this to mLocaleAwarePosition
 
   CursorOpBase* mCurrentlyRunningOp;
 
